### ISSUE TITLE
3.18 - New feature and SaaS part

### ISSUE THREAD

**Issue by piotrbak** [wp-media/wp-rocket#7070]
**User Story**
As a user, I’d like to apply the feature to the URLs visited by saas during RUCSS or OCI/LRC

**Acceptance Criteria**
- When the visits are made by SaaS (RUCSS, OCI/ALR) we need to make sure that the fonts and CSS are downloaded and the paths are rewritten

**Comment by Khadreal** [comment]
This is dependent on the completion of #7063 and #7068 

# Scope a solution: 
In the frontend subscriber class, we add a filter `rocket_performance_hints_buffer` and in the call back function we call the `rewrite_fonts` process, something like:

```
'rocket_performance_hints_buffer' => 'rewrite_fonts'
```

Then here:

https://github.com/wp-media/wp-rocket/blob/061f8b6ddbda0e4d0ee4e8c9ebfbfa5b982b12c6/inc/Engine/Media/Fonts/Context/Context.php#L19-L20

we need to remove both lines because SaaS visits the page in the following structure:
```
/?nowprocket=1&no_optimize=1&wpr_imagedimensions=1
```

Estimation
[XS]

**Comment by Khadreal** [comment]
@piotrbak should this visit happen when the feature is not enabled ?

**Comment by piotrbak** [comment]
@Khadreal This issue is not about making any visits. It's just to ensure that the feature works when making a request from SaaS

**Comment by jeawhanlee** [comment]
@Khadreal I'm not sure if we are properly covering the AC in the grooming, shouldn't we be using the `rocket_performance_hints_buffer` for this?

**Comment by jeawhanlee** [comment]
I don't think we need to check the query param since `rocket_performance_hints_buffer` will only be triggered if those query params are set. https://github.com/wp-media/wp-rocket/blob/4b9193bc96b46fd1e8a5db6f3a9bf4126a155eb5/inc/Engine/Common/PerformanceHints/Frontend/Subscriber.php#L69-L97

**Comment by Khadreal** [comment]
Yes, we don't need to, forgot to remove it after the last edit. We only need to check for the feature itself

**Comment by wordpressfan** [comment]
@jeawhanlee @Khadreal 
but `rocket_performance_hints_buffer` is working only with priority elements (performance hints) not with RUCSS also, correct?

**Comment by jeawhanlee** [comment]
@wordpressfan technically, it would work with RUCSS because we apply [wpr_imagedimensions](https://github.com/wp-media/wp-rocket/blob/4b9193bc96b46fd1e8a5db6f3a9bf4126a155eb5/inc/Engine/Common/PerformanceHints/WarmUp/Controller.php#L249) to the api.  Performance hints are enabled by default so this won't matter but a case where both performancehints features are disabled via the filter, `rocket_performance_hints_buffer` will not be set, so this grooming needs to cover that I think

**Comment by wordpressfan** [comment]
Checking that...

**Comment by wordpressfan** [comment]
So with RUCSS we always send `wpr_imagedimensions` query string (even when lrc and atf are disabled by filters).

Let's ask @wp-media/product if we need to handle the case when both lrc and atf are disabled and RUCSS is enabled to apply hosting google fonts locally?

**Comment by piotrbak** [comment]
@wordpressfan So, the situation is:
1. Both, LRC and OCI are disabled
2. RUCSS is enabled
3. Google Fonts Locally is enabled

The RUCSS should be generated with the Google Fonts being hosted locally, right?

**Comment by wordpressfan** [comment]
> The RUCSS should be generated with the Google Fonts being hosted locally, right?

Exactly, so we need to handle this case?

**Comment by MathieuLamiot** [comment]
Yes this case must be handled.

**Comment by wordpressfan** [comment]
I updated the grooming above after doing some experiments.

**Comment by jeawhanlee** [comment]
LGTM

**Comment by wordpressfan** [comment]
That was so quick :D @jeawhanlee  
I tested the following variations and it works as expected:

1. RUCSS enabled, lrc enabled, and atf enabled
2. RUCSS enabled, lrc disabled, and atf disabled
